pgmpy/inference/dbn_inference.py

```diff
@@ -0,0 +1,98 @@
+from collections import defaultdict
+from pgmpy.inference import VariableElimination
+from pgmpy.factors import TabularCPD
+from pgmpy.inference import Inference
+
+
+class DBNInference(Inference):
+
+    def query(self, variables, evidence=None, elimination_order=None):
+        variable_dict = defaultdict(list)
+        for var in variables:
+            variable_dict[var[1]].append(var[0])
+
+        # This method extracts the max value in time slices
+        timerange = max(variable_dict)
```

Comment:
A simple one liner can be `timerange = max([slice for slice in itertools.chain(variable_dict, evidence)])